datajoint/table.py

```diff
@@ -172,6 +174,16 @@ def insert(self, rows, replace=False, skip_duplicates=False, ignore_extra_fields
             warnings.warn('Use of `ignore_errors` in `insert` and `insert1` is deprecated. Use try...except... '
                           'to explicitly handle any errors', stacklevel=2)
 
+        # prohibit direct inserts into auto-populated tables
+        try:
+            allow = allow_direct_insert or self._allow_insert  # only present in AutoPopulate
```

Comment:
Can you just replace the `self._allow_insert` with `getattr(self, '_allow_insert', True)` instead of using try except block? I much prefer a logic that doesn't rely on raising exceptions when not necessary.

Reply:
agreed. Done

Reply:
Awesome